### ISSUE TITLE
raspberry pi install

### ISSUE THREAD

**Issue by AFrenchCroissant** [richardg867/WaybackProxy#1]
hello,
i would like to know how do you install this proxy on a raspberry pi 4
Thanks !

**Comment by carlni2008** [comment]
Hi i a just curious, are you trying to make this project?
https://www.youtube.com/watch?v=0OB1g8CUdbA

**Comment by AFrenchCroissant** [comment]
Yes, but a simpler version, without the lcd or the knob and change the date by the configuration page

**Comment by LuukLG** [comment]
I just went with ssh into my pi or with terminal, make sure your pi can connect to the internet.

- I just installed python3 with **sudo apt install python3**

- after that, I went into my home directory and typed:
 **wget https://github.com/richardg867/WaybackProxy/archive/refs/heads/master.zip**

- then I typed **unzip master.zip**

-  then **cd WaybackProxyt-master** to get in to the newly created directory.

- type **python3 waybackproxy.py**

- proxy is now active on the pi on default port 8888 

- search for the proxy settings on the pc you want to try and config the pi's ip address combined with port  8888.

- test with a site like google.com to see if it works as there is a default date in the config file.

**Comment by maxhayim** [comment]
> Yes, but a simpler version, without the lcd or the knob and change the date by the configuration page

I'm actually trying to make one with the dial and LCD.

**Comment by AFrenchCroissant** [comment]
> I just went with ssh into my pi or with terminal, make sure your pi can connect to the internet.
> 
> - I just installed python3 with **sudo apt install python3**
> 
> - after that, I went into my home directory and typed:
>  **wget https://github.com/richardg867/WaybackProxy/archive/refs/heads/master.zip**
> 
> - then I typed **unzip master.zip**
> 
> -  then **cd WaybackProxyt-master** to get in to the newly created directory.
> 
> - type **python3 waybackproxy.py**
> 
> - proxy is now active on the pi on default port 8888 
> 
> - Go to the proxy settings on the pc you want to use for browsing the old web and config the pi's ip address combined with port 8888.
> 
> - test with a site like microsoft.com to see if it works as this website has an archived paged around the default DATE. this parameter can be found (and edited) in the config.py file
> 

Tysm
but is there a way to start the python program on startup ?
Thanks again !

**Comment by LuukLG** [comment]
There are multiple ways. I did it with a systemd unit file. Follow Method 4: https://www.dexterindustries.com/howto/run-a-program-on-your-raspberry-pi-at-startup/ 

sudo nano /lib/systemd/system/waybackproxy.service

add this text:

 [Unit]
 Description=waybackproxy
 After=multi-user.target

 [Service]
 Type=idle
 ExecStart=/usr/bin/python3 /home/pi/**<directory of waybackproxy>**/waybackproxy.py

 [Install]
 WantedBy=multi-user.target


then edit permissions to 644 :

sudo chmod 644 /lib/systemd/system/waybackproxy.service


sudo systemctl daemon-reload
sudo systemctl enable waybackproxy.service

sudo systemctl start waybackproxy

newtstat -at 
check if there is an entry with 
Proto Recv-Q Send-Q Local Address           Foreign Address         State
tcp        0      0 0.0.0.0:8888            0.0.0.0:*               LISTEN

**Comment by LuukLG** [comment]
There are multiple ways. I did it with a systemd unit file. Follow Method 4: https://www.dexterindustries.com/howto/run-a-program-on-your-raspberry-pi-at-startup/

sudo nano /lib/systemd/system/waybackproxy.service

add this text:

[Unit]
Description=waybackproxy
After=multi-user.target

[Service]
Type=idle
ExecStart=/usr/bin/python3 /home/pi//waybackproxy.py

[Install]
WantedBy=multi-user.target

close the editor with CTRL + X and press y to save

then edit permissions to 644 :

sudo chmod 644 /lib/systemd/system/waybackproxy.service

sudo systemctl daemon-reload
sudo systemctl enable waybackproxy.service

sudo systemctl start waybackproxy

sudo netstat -at
check if there is an entry with
Proto Recv-Q Send-Q Local Address Foreign Address State
tcp 0 0 0.0.0.0:8888 0.0.0.0:* LISTEN

then reboot your pi and check if it starts on boot.

**Comment by thomwatson** [comment]
> > Yes, but a simpler version, without the lcd or the knob and change the date by the configuration page
> 
> I'm actually trying to make one with the dial and LCD.

If you manage it, I'll happily buy the project files from you. Been trying to work this out for a good day now.

**Comment by carlni2008** [comment]
Hi thanks for replying, I am also trying to make this, and thanks for the guide. Since i have the supplies, I am going to make one with a lcs screen and a knob . I'll tell you if I suceed making this.

**Comment by carlni2008** [comment]
Also, how do I connect to the proxy on windows 11 to test it? Right now it doesn't work. :(

**Comment by carlni2008** [comment]
> Also, how do I connect to the proxy on windows 11 to test it? Right now it doesn't work. I have the program running on my raspberry pi without changing the config.

I figured configing the proxy on windows only works for micosoft edge.

**Comment by Humans5235** [comment]
> > Also, how do I connect to the proxy on windows 11 to test it? Right now it doesn't work. I have the program running on my raspberry pi without changing the config.
> 
> I figured configing the proxy on windows only works for micosoft edge.

I have a similar problem to you and don't know how to fix it. I have followed all the steps listed above and set up the proxy settings. Whenever I search something in the search bar on google chrome it says 'ERR_TUNNEL_CONNECTION_FAILED', but if I type in a link to the website, e.g `apple.com` it works. In addition, whenever I successfully enter a website, I dont have a secure connection. I dont really mind if its there because I am pretty sure it is from having the proxy server.

I have tried testing somethings out in an attempt to find out whats happening. I found that if the website has '`https://www.`apple.com' Infront it does not work, but if you change that to '`http://`apple.com' it works.

It would be a pleasure if someone where to comment here and explain the issue and mainly how to fix it.
- Thanks

**Comment by LuukLG** [comment]
Just a gamble: Maybe this has something to do with modern browsers trying to use https by default when you enter an url without https or https, while sites back then mostly had http for their HomePages? 
curious if this problem also exist for old browsers on old computers.

**Comment by carlni2008** [comment]
For me, I just setted up the proxy in the settings, and it works on microsoft edge not chrome for me.

**Comment by carlni2008** [comment]
My progress so far: I suceffully connected the lcd screen LCD1602 and a Rotary Potentiometer. The Rotary Potentiometer is powered by ADC Module PCF8591. The lcd screen display the current time, and I didn't finish programming the Rotary Potentiometer yet.

**Comment by ColdSalt09** [comment]
Hey guys I have some questions
How I can change the date of the sites (I don't have a rotary thing)
Also I connect it to my PC from my raspberry pi using a lan cable?

**Comment by carlni2008** [comment]
Hi sorry I completely forgot this project. To change the dates of the sites, I think you'll have to make a program that changes the config date, like a UI of some sort of you have a display for your RPI. I just bought a cheap dial for that function.

**Comment by vanhoteen** [comment]
> 
hello I have been investigating but I don't know how to program in pyhton, if you can please, I found a lot of encoder documentation that shows options on screens but not that they modify the config.py

example 

https://gist.github.com/maxmacstn/805991009e9302977f694e5b17a62b73
